src/test/java/seedu/address/model/localcourse/UniqueLocalCourseListTest.java

```diff
@@ -0,0 +1,191 @@
+package seedu.address.model.localcourse;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalObjects.CS1231S;
+import static seedu.address.testutil.TypicalObjects.CS2030S;
+import static seedu.address.testutil.TypicalObjects.CS2040S;
+import static seedu.address.testutil.TypicalObjects.CS3230;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.localcourse.exceptions.DuplicateLocalCourseException;
+import seedu.address.model.localcourse.exceptions.LocalCourseNotFoundException;
+import seedu.address.testutil.LocalCourseBuilder;
+
+public class UniqueLocalCourseListTest {
+
+    private final UniqueLocalCourseList uniqueLocalCourseList = new UniqueLocalCourseList();
+
+    // Tests contains, add
+
+    @Test
+    public void contains_nullLocalCourse_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueLocalCourseList.contains(null));
+    }
+
+    @Test
+    public void contains_localCourseNotInList_returnsFalse() {
+        assertFalse(uniqueLocalCourseList.contains(CS2040S));
+    }
+
+    @Test
+    public void contains_localCourseInList_returnsTrue() {
+        uniqueLocalCourseList.add(CS2040S);
+        assertTrue(uniqueLocalCourseList.contains(CS2040S));
+    }
+
+    @Test
+    public void contains_localCourseWithSameIdentityFieldsInList_returnsTrue() {
+        uniqueLocalCourseList.add(CS3230);
+        LocalCourse localCourseWithSameIdentityFields =
```

Comment:
so the identity field is only local code right

Reply:
Yep.